dftimewolf/lib/collectors/grr_base.py

```diff
@@ -107,20 +104,20 @@ def _WrapGRRRequestWithApproval(
         return grr_function(*args, **kwargs)
 
       except grr_errors.AccessForbiddenError as exception:
-        self.logger.info('No valid approval found: {0!s}'.format(exception))
+        logger.info('No valid approval found: {0!s}'.format(exception))
         # If approval was already sent, just wait a bit more.
         if approval_sent:
-          self.logger.info('Approval not yet granted, waiting {0:d}s'.format(
+          logger.info('Approval not yet granted, waiting {0:d}s'.format(
               self._CHECK_APPROVAL_INTERVAL_SEC))
-          self.logger.success(approval_url)
+          logger.success(approval_url) # type: ignore
           time.sleep(self._CHECK_APPROVAL_INTERVAL_SEC)
           continue
 
         # If no approvers were specified, abort.
         if not self.approvers:
           message = ('GRR needs approval but no approvers specified '
                      '(hint: use --approvers)')
-          self.ModuleError(message, critical=True)
+          raise GrrError(message, critical=True) from exception
```

Comment:
Why not simply raise a `dfTimewolfError` here? You're not catching any `GrrError`s anyways.

Reply:
`GrrError` is a subclass of `DFTimewolfError`. Added in `Raises` notations.

Reply:
I meant, why introduce a new `GrrError` class, when dfTimewolf error seems to be enough for this purpose? I don't see anywhere in the code where you're doing `except GrrError as error` (i.e. catching that specific error type)

Reply:
Good point. Removed `GrrError` in favour of `DFTimewolfError`.